docs/language/ql-handbook/language.rst

```diff
@@ -42,6 +42,79 @@ A QL program can be *evaluated* (see `Evaluation <#evaluation>`__) to produce a
 
 For a QL program to be *valid*, it must conform to a variety of conditions that are described throughout this specification; otherwise the program is said to be *invalid*. An implementation of QL must detect all invalid programs and refuse to evaluate them.
 
+Library path
+------------
+
+The library path an ordered list of directory locations that is used
+in for resolving module imports, described below.  It is not strictly
+speaking a core part of the QL language, since different
+implementations of QL construct it in slightly different ways. Most QL
+tooling also allows specifying it explicitly on the command line for a
+particular invocation, though that is that is rarely done, and only
+useful in very special situation. This section describes the default
+construction of the library path.
+
+First, determine the *query directory* of the ``.ql`` file being
+compiled. Starting with the directory containing the ``.ql`` file, and
+walking up the directory structure, each directory is checked for a
+file called ``queries.xml`` or ``qlpack.yml``. The first directory
+where such a file is found is the query directory. If there is no such
+directory, the directory of the ``.ql`` file itself is the query
+directory.
+
+A ``queries.xml`` file that defines a query directory must always
+contain, containing a single top-level tag named
+``queries``, which has a ``language`` attribute set to the identifier
+of the active database scheme (for example, ``<queries
+language="java"/>``).
+
+A ``qlpack.yml`` file defines a `QL pack
+<https://help.semmle.com/codeql/codeql-cli/reference/qlpack-overview.html>`;
+its content is described in the CodeQL CLI documentation. This file
+will not be recognized when using older QL tooling that is not based
```

Comment:
```suggestion
will not be recognized when using legacy tools that are not based
```

Using "legacy tools" here for consistency with other parts of the documentation, e.g. [Notes for legacy QL CLI users](https://help.semmle.com/codeql/notes-ql-users.html).